src/services/identity/CollaboratorsService.ts

```diff
@@ -225,30 +231,51 @@ class CollaboratorsService {
     siteName: string,
     userId: string
   ): Promise<CollaboratorRoles | null> => {
-    const site = await this.siteRepository.findOne({
-      include: [
-        {
-          model: User,
-          as: "site_members",
-          where: {
-            id: userId,
-          },
-        },
-        {
-          model: Repo,
-          where: {
-            name: siteName,
-          },
-        },
-      ],
-    })
+    // Note: We could make the query even cheaper if we assume the site and user cannot be deleted when this verification is done
+    // Basically with just:
+    // SELECT sm.role
+    // FROM
+    //     site_members sm,
+    //     repos r,
+    // WHERE
+    //     r.site_id = sm.site_id
+    //     AND r.name = :siteName
+    //     AND sm.user_id = :userId
+    //     AND r.deleted_at IS NULL
+
+    const records = (await this.sequelize.query(
+      `
+        SELECT sm.role
+        FROM
+            site_members sm,
```

Comment:
nit: not a fan of these renames rather keep them as is for readability

Reply:
Sure, I can drop the aliases 🤷 . For short queries, I think readability is still OK, but it's your codebase! 😄

Reply:
Just to drop it here in fact. Personally I find that when the table name is longer than the field names, the query reads worse 😰

Reply:
Done! Dropped the short aliases: https://github.com/isomerpages/isomercms-backend/pull/1332/commits/43621ae20891d872630d06aa08138699b9e28ac1